### PR TITLE
Adds tests, CLI, and fixes some tiny bugs.

### PR DESCRIPTION
**Pull request by gguuss**

### CODE REVIEW

File: iot/api-client/manager/manager.py [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+
+    return topic
+
+
+def get_client(service_account_json, api_key):
+    """Returns an authorized API client by discovering the IoT API using the
+    provided API key and creating a service object using the service account
+    credentials JSON."""
+    # [START authorize]
+    API_SCOPES = ['https://www.googleapis.com/auth/cloud-platform']
```
Comment (jonparrott) [inline iot/api-client/manager/manager.py]: you can just use `lower_snake_case` for this stuff.

File: iot/api-client/manager/manager.py [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+    API_VERSION = 'v1beta1'
+    DISCOVERY_API = 'https://cloudiot.googleapis.com/$discovery/rest'
+    SERVICE_NAME = 'cloudiotcore'
+
+    credentials = service_account.Credentials.from_service_account_file(
+            service_account_json)
+    scoped_credentials = credentials.with_scopes(API_SCOPES)
+
+    if not credentials:
```
Comment (jonparrott) [inline iot/api-client/manager/manager.py]: Did you not like the error message from `form_service_account_file`?

File: iot/api-client/manager/manager.py [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+        'id': device_id,
+        'credentials': [{
+            'publicKey': {
+                'format': 'RSA_X509_PEM',
+                'key': certificate
+            }
+        }]
+    }
+
+    return client.projects().locations().registries().devices(
```
Comment (jonparrott) [inline iot/api-client/manager/manager.py]: I generally like to break these up like this:


```python
devices = client.projects().locations().registries().devices()
return devices.create(...).execute()
```

File: iot/api-client/manager/manager.py [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+        device_id):
+    """Retrieve the device with the given id."""
+    # [START delete_device]
+    print('Getting device')
+    client = get_client(service_account_json, api_key)
+    registry_name = 'projects/{}/locations/{}/registries/{}'.format(
+            project_id, cloud_region, registry_id)
+
+    device_name = '{}/devices/{}'.format(registry_name, device_id)
+    device = client.projects().locations().registries().devices(
```
Comment (jonparrott) [inline iot/api-client/manager/manager.py]: recommend that same splitting technique here.

File: iot/api-client/manager/manager.py [hunk 5]
```diff
@@ -0,0 +1,472 @@
...
+def patch_es256_auth(
+        service_account_json, api_key, project_id, cloud_region, registry_id,
+        device_id, public_key_file):
+    """Patch the device to add an ES256 public key to the device."""
+    print('Patch device with ES256 certificate')
+    client = get_client(service_account_json, api_key)
+    registry_path = 'projects/{}/locations/{}/registries/{}'.format(
+            project_id, cloud_region, registry_id)
+
+    with open(public_key_file) as f:
```
Comment (jonparrott) [inline iot/api-client/manager/manager.py]: use `io.open`, please.

**Comment by gguuss** [comment]
Done

**Comment by jonparrott** [comment]
@dpebot will you merge when travis passes plz?

**Pull request merged**